core/src/main/java/aima/core/search/basic/uninformedsearch/BreadthFirstSearch.java

```diff
@@ -10,31 +10,79 @@
 import java.util.LinkedList;
 import java.util.Queue;
 
+/**
+ * <pre>
+ *  function BREADTH-FIRST-SEARCH(problem) returns a solution, or failure
+ *  if problem's initial state is a goal then return empty path to initial state
+ *  frontier ← a FIFO queue initially containing one path, for the problem's initial state
+ *  reached ← a set of states; initially empty
+ *  solution ← failure
+ *  while frontier is not empty do
+ *    parent ← the first node in frontier
+ *    for child in successors(parent) do
+ *      s ← child.state
+ *      if s is a goal then
+ *        return child
+ *      if s is not in reached then
+ *        add s to reached
+ *        add child to the end of frontier
+ *  return solution
+ * </pre>
+ *
+ * Figure 3.9 Breadth-first search algorithm.
+ *
+ * @param <A> The generic object representing actions.
+ *
+ * @param <S> The generic object representing state.
+ *
+ * @author samagra
+ */
 public class BreadthFirstSearch<A,S> implements GenericSearchInterface<A,S> {
+    // Node factory is just a helper class to generate new nodes
+    // It takes the problem and current state in order to generate new nodes
     NodeFactory<A,S> nodeFactory = new BasicNodeFactory();
 
+    /**
+     * function BREADTH-FIRST-SEARCH(problem) returns a solution, or failure
+     * @param problem The search problem.
+     * @return The goal node if any else null.
+     */
     @Override
     public Node<A,S> search(Problem<A,S> problem) {
+        //  if problem's initial state is a goal then return empty path to initial state
         if (problem.isGoalState(problem.initialState())){
             return nodeFactory.newRootNode(problem.initialState());
         }
+        // frontier ← a FIFO queue initially containing one path, for the problem's initial state
         Queue<Node> frontier = new LinkedList<>();
         ((LinkedList<Node>) frontier).add(nodeFactory.newRootNode(problem.initialState()));
+        // reached ← a set of states; initially empty
         HashSet<S> reached = new HashSet<>();
+        // solution ← failure
```

Comment:
I don't like this style, where the comments duplicate the code. I think it is good to have the pseudocode in a block comment, but then the Java implementation should stand by itself. Any Java programmer knows what `reached = new HashSet<>();` means; we don't need to duplicate that in the comments.

Reply:
@norvig 
Sure, sir, I will have these duplicate comments removed. I added them just because of the older versions. I will have them removed.